test/decimal.test.js

```diff
@@ -63,4 +69,431 @@ describe('model with decimal property', function() {
+
+      modelWithDeepNestedDecimalProps.create(createData)
+        .then(function(inst) {
+          instanceId = inst.id;
+          return findModelInstanceAsync('modelWithDeepNestedDecimalProps', instanceId);
+        })
+        .then(function(createdInstance) {
+          createdInstance.imdb.rating.should.be.instanceOf(Decimal128);
+          createdInstance.imdb.rating.should.deepEqual(Decimal128.fromString('4.5'));
+          createdInstance.imdb.innerObj.innerObj.innerObj.innerObj.testDecimal
+            .should.be.instanceOf(Decimal128);
+          createdInstance.imdb.innerObj.innerObj.innerObj.innerObj.testDecimal
+            .should.deepEqual(Decimal128.fromString('55.55'));
+          createdInstance.imdb.innerArray[0].testDecimal
+            .should.be.instanceOf(Decimal128);
+          createdInstance.imdb.innerArray[0].testDecimal
+            .should.deepEqual(Decimal128.fromString('99.55'));
+          createdInstance.imdb.innerArray[0].ObjInsideAnArray.testDecimal
+            .should.deepEqual(Decimal128.fromString('99.55'));
+          createdInstance.imdb.innerArray[0].ObjInsideAnArray.testDecimal
+            .should.be.instanceOf(Decimal128);
+          createdInstance.imdb.innerArray[0].nestedArray[0]
+            .testDecimal.should.be.instanceOf(Decimal128);
+          createdInstance.imdb.innerArray[0].nestedArray[0]
+            .testDecimal.should.deepEqual(Decimal128.fromString('99.55'));
+          return modelWithDeepNestedDecimalProps.updateAll({id: instanceId}, updateData);
+        })
+        .then(function() {
+          return findModelInstanceAsync('modelWithDeepNestedDecimalProps', instanceId);
+        })
+        .then(function(updatedInstance) {
+          updatedInstance.imdb.rating.should.be.instanceOf(Decimal128);
+          updatedInstance.imdb.rating.should.deepEqual(Decimal128.fromString('7.5'));
+          updatedInstance.imdb.innerObj.innerObj.innerObj.innerObj.testDecimal
+            .should.be.instanceOf(Decimal128);
+          updatedInstance.imdb.innerObj.innerObj.innerObj.innerObj.testDecimal
+            .should.deepEqual(Decimal128.fromString('77.77'));
+          updatedInstance.imdb.innerArray[0].testDecimal
+            .should.be.instanceOf(Decimal128);
+          updatedInstance.imdb.innerArray[0].testDecimal
+            .should.deepEqual(Decimal128.fromString('11.11'));
+          updatedInstance.imdb.innerArray[0].ObjInsideAnArray.testDecimal
+            .should.deepEqual(Decimal128.fromString('77.77'));
+          updatedInstance.imdb.innerArray[0].ObjInsideAnArray.testDecimal
+            .should.be.instanceOf(Decimal128);
+          updatedInstance.imdb.innerArray[0].nestedArray[0]
+            .testDecimal.should.be.instanceOf(Decimal128);
+          updatedInstance.imdb.innerArray[0].nestedArray[0]
+            .testDecimal.should.deepEqual(Decimal128.fromString('77.77'));
+          done();
+        });
+    });
+
+    function findModelInstance(modelName, id, cb) {
```

Comment:
IIUC, this method is not returning an instance of the model class, just the raw model data. In which case I am proposing a slightly different name:

```suggestion
    function findRawModelData(modelName, id, cb) {
```

Similarly for `findModelInstanceAsync` below.